core/trino-main/src/main/java/io/trino/operator/HttpPageBufferClient.java

```diff
@@ -724,6 +747,15 @@ private static boolean getComplete(Response response, URI uri)
             return Boolean.parseBoolean(bufferComplete);
         }
 
+        private static boolean getTaskFailed(Response response, URI uri)
+        {
+            String taskFailed = response.getHeader(TRINO_TASK_FAILED);
```

Comment:
Why do we send this header on every response instead of just when the task actually fails?

Reply:
I just followed the pattern used for other headers (e.g.: the `TRINO_BUFFER_COMPLETE` header is also set every time, even when the buffer is not complete)